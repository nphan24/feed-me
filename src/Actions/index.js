export const postRecipes = recipes => ({
  type: 'POST_RECIPES',
  recipes
});

export const replaceRecipes = recipes => ({
  type: 'REPLACE_RECIPES',
  recipes
});

export const addFavorite = recipe => ({
  type: 'ADD_FAVORITE',
  recipe
});

export const removeFavorite = id => ({
  type: 'REMOVE_FAVORITE',
  id
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const promptSignIn = error => ({
  type: 'PROMPT_SIGNIN',
  error
});