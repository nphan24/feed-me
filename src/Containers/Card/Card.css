.recipe-card {
  border: 1px solid black;
  width: 280px;
  height: 340px;
  margin: 5px;
}

.favorite-button {
  display: block;
  margin: 0 auto 10px auto;
  border-radius: 30px 30px 30px 30px;
  background-color: rgb(24, 204, 33)
}

.recipe-image {
  width: 150px;
  height: 150px;
}

