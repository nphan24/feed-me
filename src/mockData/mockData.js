/* eslint-disable */

export const mockRecipe = {
  name: 'pasta',
  image: 'url',
  totalTime: '30',
  id: 2
};

export const uncleanedMockRecipeData = { matches: [
  { recipeName: 'tofu', 
    imageUrlsBySize: 'url', 
    ingredients: ['',''], 
    totalTimeInSeconds: 3600, 
    id: 2,
    pasta: 'speghetti',
    fruit: 'mango'
  }
]
 };

export const cleanedMockRecipeData = [{ 
  Ingredients: ['', ''], 
  id: 2, 
  image: undefined, 
  name: 'tofu', 
  totaltime: 60 
}];

export const mockRecipesArray = [
  {
    name: 'pasta',
    image: 'url',
    totalTime: '30',
    id: 2
  },
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];

export const mockRemainingArray = [
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];
