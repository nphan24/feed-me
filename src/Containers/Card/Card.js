import React from 'react';
import { connect } from 'react-redux';
import * as Actions from '../../Actions/';
import './Card.css';

export const Card = ({
  recipe, 
  favorites, 
  addFavorite, 
  removeFavorite 
}) => {
  const { name, image, totalTime, source } = recipe;
  const cardClass = favorites.find(fav => fav.name === recipe.name) ? 'selected' : '';
  
  const toggleFavorite = (recipe) => {
    if (!favorites.find(fav => fav.id === recipe.id)) {
      addFavorite(recipe);
    } else {
      removeFavorite(recipe.id);
    }
  };

  return (
    <div className='recipe-card'>
      <p>{name}</p>
      <button 
        className={`favorite-button ${cardClass}`}
        onClick={()=> toggleFavorite(recipe)}>
        <span role='img' aria-labelledby='favorite star' className='favorites-image'>
        &#x2B50;
        </span>
      </button>
      <img src={image} className="recipe-image" alt='recipe'/>
      <p>Total cook time: {totalTime} mins</p>
      <a href={source} target="..blank" className='link-instructions'>
        Make This!
      </a>
    </div>
  );
};

export const mapStateToProps = state => ({ 
  favorites: state.favorites 
});

export const mapDispatchToProps = dispatch => ({
  addFavorite: recipe => dispatch(Actions.addFavorite(recipe)),
  removeFavorite: id => dispatch(Actions.removeFavorite(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(Card);



