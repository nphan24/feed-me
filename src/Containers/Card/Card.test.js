import React from 'react';
import { shallow } from 'enzyme';
import { Card, mapStateToProps, mapDispatchToProps } from './Card';
import * as mock from '../../mockData/mockData';
import * as Actions from '../../Actions';

describe('Card', () => {
  let wrapper;
  let mockRecipe = mock.mockRecipe;
  let mockaddFavorite = jest.fn();
  let mockremoveFavorite = jest.fn();
  let mockFavoritesToAdd = [{ 
    name: 'pasta', 
    image: 'url', 
    totalTime: '30', 
    id: 2 
  }];
  let mockFavorites = [{ 
    name: 'pasta', 
    image: 'url', 
    totalTime: '30', 
    id: 4 
  }];

  beforeEach(()=> {
    wrapper = shallow(<Card 
      recipe={mockRecipe}
      favorites={mockFavoritesToAdd}
      removeFavorite={mockremoveFavorite}
      addFavorite={mockaddFavorite}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call removeFavorite in toggleFavorite if the ids match', () => {
    wrapper.find('button').simulate('click');
    expect(mockremoveFavorite).toHaveBeenCalledWith(2);
  });

  it('should call addFavorite in toggleFavorite if there are no matches', () => {
    wrapper = shallow(<Card 
      recipe={mockRecipe} 
      favorites={mockFavorites} 
      removeFavorite={mockremoveFavorite} 
      addFavorite={mockaddFavorite} 
    />);
    wrapper.find('button').simulate('click');
    expect(mockaddFavorite).toHaveBeenCalled();
  });
});

describe('mapStateToProps', () => {
  it('should map favorites to props', () => {
    const state = {favorites: mock.favorites};
    const expected = mock.favorites;
    const mapped = mapStateToProps(state);
    expect(mapped.favorites).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with the correct params on addFavorites', () => {
    const mockDispatch = jest.fn();
    const recipe = mock.mockRecipe;
    const expected = Actions.addFavorite(recipe);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addFavorite(recipe);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with the correct params on removeFavorites', () => {
    const mockDispatch = jest.fn();
    const id = 24;
    const expected = Actions.removeFavorite(id);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.removeFavorite(id);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});