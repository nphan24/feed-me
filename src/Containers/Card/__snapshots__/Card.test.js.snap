// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addFavorite={[Function]}
    favorites={
        Array [
            Object {
              "id": 2,
              "image": "url",
              "name": "pasta",
              "totalTime": "30",
            },
          ]
    }
    recipe={
        Object {
            "id": 2,
            "image": "url",
            "name": "pasta",
            "totalTime": "30",
          }
    }
    removeFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-button selected"
          onClick={[Function]}
>
          <span
                    aria-labelledby="favorite star"
                    className="favorites-image"
                    role="img"
          >
                    ⭐
          </span>
</button>,
        <img
          alt="recipe"
          className="recipe-image"
          src="url"
/>,
        <p
          className="recipe-name"
>
          pasta
</p>,
        <p
          className="recipe-cooktime"
>
          Total cook time: 
           mins
</p>,
        <a
          className="link-instructions"
          href={undefined}
          target="..blank"
>
          Make This!
</a>,
      ],
      "className": "recipe-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            aria-labelledby="favorite star"
            className="favorites-image"
            role="img"
>
            ⭐
</span>,
          "className": "favorite-button selected",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-labelledby": "favorite star",
            "children": "⭐",
            "className": "favorites-image",
            "role": "img",
          },
          "ref": null,
          "rendered": "⭐",
          "type": "span",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "recipe",
          "className": "recipe-image",
          "src": "url",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "pasta",
          "className": "recipe-name",
        },
        "ref": null,
        "rendered": "pasta",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Total cook time: ",
            undefined,
            " mins",
          ],
          "className": "recipe-cooktime",
        },
        "ref": null,
        "rendered": Array [
          "Total cook time: ",
          undefined,
          " mins",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Make This!",
          "className": "link-instructions",
          "href": undefined,
          "target": "..blank",
        },
        "ref": null,
        "rendered": "Make This!",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-button selected"
            onClick={[Function]}
>
            <span
                        aria-labelledby="favorite star"
                        className="favorites-image"
                        role="img"
            >
                        ⭐
            </span>
</button>,
          <img
            alt="recipe"
            className="recipe-image"
            src="url"
/>,
          <p
            className="recipe-name"
>
            pasta
</p>,
          <p
            className="recipe-cooktime"
>
            Total cook time: 
             mins
</p>,
          <a
            className="link-instructions"
            href={undefined}
            target="..blank"
>
            Make This!
</a>,
        ],
        "className": "recipe-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              aria-labelledby="favorite star"
              className="favorites-image"
              role="img"
>
              ⭐
</span>,
            "className": "favorite-button selected",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-labelledby": "favorite star",
              "children": "⭐",
              "className": "favorites-image",
              "role": "img",
            },
            "ref": null,
            "rendered": "⭐",
            "type": "span",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "recipe",
            "className": "recipe-image",
            "src": "url",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "pasta",
            "className": "recipe-name",
          },
          "ref": null,
          "rendered": "pasta",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Total cook time: ",
              undefined,
              " mins",
            ],
            "className": "recipe-cooktime",
          },
          "ref": null,
          "rendered": Array [
            "Total cook time: ",
            undefined,
            " mins",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Make This!",
            "className": "link-instructions",
            "href": undefined,
            "target": "..blank",
          },
          "ref": null,
          "rendered": "Make This!",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
